src/BackgroundTask/SingleScheduledBackgroundTask.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Abstract class SingleScheduledBackgroundTask.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\BackgroundTask;
+
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+
+/**
+ * Abstract base class for using cron to execute a background task that runs only once.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.1
+ * @internal
+ */
+abstract class SingleScheduledBackgroundTask implements Service, Registerable, Conditional {
```

Comment:
I made a brief attempt at this, but it became complicated when it came to `MonitorCssTransientCaching`; I was going to have to make modifications to that class to accommodate the signature changes to `CronBasedBackgroundTask`. I think we should leave this alone for now. Using class extension might be good for DRY, but cron tasks are already difficult to debug as it is, so it might be wise to keep these classes straightforward, even if some code is repeated. I'm happy to revisit, though, if others disagree.

Reply:
Note that the initial abstraction was more flexible, you can still see it here: https://github.com/ampproject/amp-wp/pull/4177/commits/4d2bdba6e40c7d6738bb9f3afb71c4b66cef1d1d

It was later simplified, as that kind of abstraction was not really needed, with the intention of adapting it once there is an actual need.